.github/workflows/vscode.yml

```diff
@@ -14,30 +14,39 @@ on:
 jobs:
   test:
     strategy:
+      fail-fast: false
       matrix:
-        # TODO: support other platform such as macos
-        os: [ubuntu-latest]
+        os: [ubuntu-latest, macos-latest, windows-latest]
         ruby: ['3.1.0']
     runs-on: ${{ matrix.os }}
     steps:
     - name: Checkout
       uses: actions/checkout@v3
     - name: Set up Ruby
+      id: setup-ruby
       uses: ruby/setup-ruby@v1
       with:
         ruby-version: ${{ matrix.ruby }}
+        bundler: latest
     - name: Install Node.js
       uses: actions/setup-node@v3
       with:
         node-version: 16.x
-    - run: gem install bundler
-    - run: npm install
+        cache: npm
+    - name: Install npm packages
+      run: npm ci
```

Comment:
[note] [`npm ci`](https://docs.npmjs.com/cli/v9/commands/npm-ci) respects `package-lock.json`. This should be preferable on CI.